### PR TITLE
[REVIEW] Add Databricks sample notebook

### PR DESCRIPTION
**Pull request by Nanthini10**
Adding
- Databricks notebook for MLflow + Hyperopt +RAPIDS demo

- Data acquisition notebook - pulls from S3 bucket and places on DBFS

### CODE REVIEW

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+    "      \n",
+    "### MLFlow\n",
+    "\n",
+    "&nbsp;&nbsp;&nbsp;&nbsp;[MLflow](https://mlflow.org/) is used for managing the machine learning lifecycle. It provides a way to track experiements and store data about them. MLflow also provides a way to deploy ML models. We'll make use of MLflow to store information about the models and use the built-in integration of MLflow into Databricks to register and store models.\n",
+    "\n",
+    "### Hyperopt\n",
+    "\n",
+    "[Hyperopt](http://hyperopt.github.io/hyperopt/) is a library for finding the best hyperparameters for a given objective function. It provides a way to choose the objective function, the search algorithm, the database in which to store the eval points. We'll use this to define the parameter space to search over. \n",
+    "\n",
+    "We'll use airlines data for this demo - the aim is to predict if an airline will be delayed or not using a target variable `ArrDelayBinary`. We'll use a Random Forest Classifier as the model for the learning task. \n",
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:21]: Probably "We'll use FAA flight history data for this demo - the aim is to predict if a given flight ..."

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+    "      \n",
+    "### MLFlow\n",
+    "\n",
+    "&nbsp;&nbsp;&nbsp;&nbsp;[MLflow](https://mlflow.org/) is used for managing the machine learning lifecycle. It provides a way to track experiements and store data about them. MLflow also provides a way to deploy ML models. We'll make use of MLflow to store information about the models and use the built-in integration of MLflow into Databricks to register and store models.\n",
+    "\n",
+    "### Hyperopt\n",
+    "\n",
+    "[Hyperopt](http://hyperopt.github.io/hyperopt/) is a library for finding the best hyperparameters for a given objective function. It provides a way to choose the objective function, the search algorithm, the database in which to store the eval points. We'll use this to define the parameter space to search over. \n",
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:19]: Indent is inconsistent between paragraphs. Probably no indent for any?

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    "import sklearn.model_selection"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preapring the data\n",
+    "\n",
+    "1. Download the data into your local system from [Airline 20M](https://rapids-csp.s3-us-west-2.amazonaws.com/data/airline_20000000.orc). This file is in ORC format, since Pandas can't easily read ORC files, we will convert the file to Parquet. \n",
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:64]: Could we just share the .parquet file from this bucket?
Also, can people execute a python command to pull the data onto dbfs without having to go through a local system?

Comment (JohnZed) [reply databricks/notebooks/rapids_airline_hyperopt.ipynb:64]: Maybe just use urllib to grab from that bucket?

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from contextlib import contextmanager\n",
+    "import time\n",
+    "\n",
+    "@contextmanager\n",
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:95]: Maybe document:
`# Small utility that times a block of code and prints how long it took to execute`

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 5]
```diff
@@ -0,0 +1,419 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Experiment set-up\n",
+    "Here we'll define the input file path, max evaluation runs and parallelism. \n",
+    "\n",
+    "Note: For using `MAX_PARALLEL` on DataBricks, make sure to set up the cluster with minimum number of nodes as `MAX_PARALLEL`"
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:110]: Maybe rephrase a little: `To use MAX_PARALLEL on DataBricks, make sure your cluster has at least MAX_PARALLEL nodes`.

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 6]
```diff
@@ -0,0 +1,419 @@
...
+    "        mod = sklearn.ensemble.RandomForestClassifier(max_depth=max_depth,\n",
+    "                                             max_features=max_features,\n",
+    "                                             n_estimators=n_estimators,\n",
+    "                                             n_jobs=-1) # Use all available CPUs\n",
+    "        mod.fit(X_train, y_train)\n",
+    "\n",
+    "    mlflow.sklearn.log_model(mod, \"RF_model_cpu_large_\",\n",
+    "                           registered_model_name=registered_model_name)\n",
+    "\n",
+    "    with timed(\"predict\"):\n",
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:195]: The indentation seems wrong here?

File: databricks/notebooks/rapids_airline_hyperopt.ipynb [hunk 7]
```diff
@@ -0,0 +1,419 @@
...
+    "mlflow.end_run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Results\n",
+    "\n",
+    "On a `p3.2xlarge` cluster on Databricks, the runtimes observed were 24 minutes for GPU and nearly 14 hours for CPU. We illustrate that RAPIDS on GPU can give up to 35x speedups. This makes a huge difference when deciding whether or not to run a Hyperparamter optimzation.  "
```
Comment (JohnZed) [inline databricks/notebooks/rapids_airline_hyperopt.ipynb:385]: I love the numbers! Maybe we can rephrase the last sentence to: "Hopefully it will be easier to integrate hyperparameter optimization into your workflow if it can run in a coffee break rather than running overnight!"

(Also, you have a trailing blank cell at the end of the notebook that could be deleted)

File: databricks/notebooks/rapids_airline_hyperopt.ipynb (hunk 3 above)
Comment (Nanthini10) [reply databricks/notebooks/rapids_airline_hyperopt.ipynb:64]: The bucket doesn't have the parquet version of this file. And I don't have permissions to upload it. Do I ask Miro to do it?

**Comment by Nanthini10** [comment]
Tagging @drobison00 for a review. Can't seem to request review through the option
